indico/modules/events/api.py

```diff
@@ -214,11 +220,31 @@ def _addOccurrences(self, fossil, obj, startDT, endDT):
         return fossil
 
     def _makeFossil(self, obj, iface):
+        legacy_obj = obj.as_legacy if isinstance(obj, Event) else obj
         return fossilize(obj, iface, tz=self._tz, naiveTZ=self._serverTZ,
                          filters={'access': self._userAccessFilter},
-                         canModify=obj.canModify(self._aw),
+                         canModify=legacy_obj.canModify(self._aw),
                          mapClassType={'AcceptedContribution': 'Contribution'})
 
+    def _get_query_options(self, detail_level=None):
```

Comment:
ah, I thought we'd have a default value that's not None in case nothing is specified. Anyway, you always pass one here so it's a bit pointless to have a default value in my opinion